### ISSUE TITLE
BUG: Inocrrect unread updates when feed has non-unique GUIDs

### ISSUE THREAD

**Issue by ultimatespirit** [brief-rss/brief#428]
Potentially related to #178, proposed solution there however does not fix this issue.

If a feed contains multiple items with the same GUID value Brief will not properly detect that it has already "seen" that entry before and mark both items as new, unread messages. Additionally if the items are dated the newer item will be marked as the older item's date, so the GUID assumption ignores the actually marked date, and the newer item will be **duplicated** every update. I have not tested if the duplication is seen due to the included date or is due to the newer item simply being the first item sequentially in the file with that same GUID.

While yes this may be behaviour triggered by a non-standard RSS situation, as GUID normally should be, well, unique, the handling of the situation is less than ideal. Duplication of entries leads to a large ballooning in the reported unread items as well as erroneous notifications from brief about "new" updates.

A possible temporary fix to the ballooning situation could be to store a hash of the retrieved feeds files and compare hashes each update, if the hashes match don't bother actually parsing the file or updating anything. This would only work between updates to the actual feed, at which point the older items would once again be marked as unread, but it would at least solve ballooning as well as issue #178 (with same caveats).

Alternatively the whole last used feeds file may be retained and the new feeds diff'd with each other which should completely remove this problem for the lifetime of the duplicated items. Currently Brief appears to re-parse and update feeds regardless of changes or lack thereof to the actual feeds themselves.

Here is a minimal example setup to reproduce this bug, if needed I can link the real world feed that actually exhibits this behaviour.

```rss
<?xml version="1.0"?>

<rss version="2.0">
<channel>
  <title>Bug Test</title>
  <description>Some text here</description>
  <link>http://example.com/</link> 

<item>
    <title>Example 2</title>
    <description>This entry should be duplicated each update</description>
    <link>http://example.com/</link>
    <guid>http://example.com/</guid>
    <pubDate>Mon,January 21 2019 00:00:00 GMT</pubDate>
</item>

<item>
    <title>Example 1</title>
    <description>This entry should be marked as unread but NOT duplicated</description>
    <link>http://example.com/</link>
    <guid>http://example.com/</guid>
    <pubDate>Mon,January 20 2019 00:00:00 GMT</pubDate>
</item>

</channel>

</rss>
```

**Comment by Cimbali** [comment]
@tanriol I think this should be closed as its duplicate #344 has been closed. Would love to see the fix released (just ran into another feed with a duplicated post…), I hope you have the time to get around to it soon. :+1:

**Comment by tanriol** [comment]
Think so.
